### DISCUSSION TITLE
Problem getting my own test-module installed

### DISCUSSION THREAD

**Discussion by exeqtwasnotfree** [atrocore/atrocore#294] [General]
Hello.
I have made my own variant of "example-module" and put it in my own github as private.
I have tried to follow [Module Installation](https://github.com/atrocore/docs/blob/83c1c33eb770987b67ed0e31e69785afda26e926/atrocore/developer-guide/creating-own-module.md#module-installation) with some adjustment on my own module to "make it unique".

But when i add repository-block in composer.json i can see the module in Administrator -> System -> update....
But starting an update generates:

`[RuntimeException]
/var/www/.composer/cache/vcs does not exist and could not be created.  `

I have even created the directory with necessary permissions, but no.

My feeling is i made a typo somewhere, or am i wrong in that the block "repository" has to be added in composer.json in the root folder on the server, and my module added in the "require"?

My composer.json look like this:

> {
>   "name": "atrocore/skeleton-pim-no-demo",
>   "description": "Skeleton for AtroPIM",
>   "homepage": "https://atropim.com",
>   "minimum-stability": "stable",
>   "require": {
>     "atrocore/core": "*",
>     "atrocore/pim": "*",
>     "atrocore/dam": "*",
>     "atrocore/export-feeds": "*",
>     "atrocore/import-feeds": "*",
>     "MYGITHUB/MYTESTMODULE": "*"
>   },
> 
>   "repositories": [ 
>   {
>     "type": "git",
>     "url": "https://MYUSERNAME:MYTOKEN@github.com/MYGITHUB/MYTESTMODULE"
>   } 
>   ]
> }

**Comment by rratsun** [comment]
Hello,

```[RuntimeException] /var/www/.composer/cache/vcs does not exist and could not be created.``` - this because of server-user permissions.

**Reply by exeqtwasnotfree** [reply]
Hello again, I was focused on the directory the project exist in, so I didn´t notice that the directory requested was outside this folder.
Now when i created the folders in the right place and with correct permissions, it installed the module.
But now I cant find it, it don´t show on the page "Update & Modules".
I can see the module if i run "php composer.phar show" in terminal.
I continue to troubleshot, but if anyone have any input, it´s mutch appreciated.

**Reply by rratsun** [reply]
Please, check is your module has extra part in composer.json

![image](https://user-images.githubusercontent.com/33658723/161568235-03d154d9-7b23-4499-9e1a-f4da15d61c26.png)

**Reply by exeqtwasnotfree** [reply]
Yes, I have this part.
I have changed "ExampleModule" or "example-module" to my own name in this file and everywhere else needed in my copy of the module.
Maybe I have missed one place or made one correction to mutch?
I will go true my code again and double check.
